src/index.js

```diff
@@ -0,0 +1,10 @@
+const getWilayaList = require('./api/getWilayaList');
+
+const data = require('../data/WilayaList.json');
+
+const _getData = () => ([...data]);
+
+
+module.exports = {
+  getWilayaList: getWilayaList(_getData())
```

Comment:
I went for this solution of passing the dataset-clone to the internal functions to make testing easier. Calling this clone function will prevent us from having weird results. 

We can discuss whether to keep it like this or force cloning before usage inside the functions (for sure we should ensure mutating this parameter anyway)

Reply:
that's right, in this complex dataset mutation is a big problem, well thought 👌🏼